flink-connector-aws/flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/sink/DynamoDbTypeInformedElementConverter.java

```diff
@@ -0,0 +1,380 @@
+                    .build();
+        } catch (ClassCastException | IllegalArgumentException e) {
+            throw new FlinkRuntimeException(
+                    String.format(
+                            "Failed to convert %s to DynamoDbWriteRequest using %s",
+                            input, typeInfo),
+                    e);
+        }
+    }
+
+    private <attrT> TableSchema<attrT> createTableSchema(CompositeType<attrT> typeInfo)
+            throws IntrospectionException {
+        if (typeInfo instanceof RowTypeInfo) {
+            return (TableSchema<attrT>) createTableSchemaFromRowType((RowTypeInfo) typeInfo);
+        } else if (typeInfo instanceof PojoTypeInfo<?>) {
+            return createTableSchemaFromPojo((PojoTypeInfo<attrT>) typeInfo);
+        } else if (typeInfo instanceof TupleTypeInfo<?>) {
+            return createTableSchemaFromTuple((TupleTypeInfo<?>) typeInfo);
+        } else {
+            throw new IllegalArgumentException(String.format("Unsupported TypeInfo %s", typeInfo));
+        }
+    }
+
+    private TableSchema<Row> createTableSchemaFromRowType(RowTypeInfo typeInfo) {
+        StaticTableSchema.Builder<Row> tableSchemaBuilder =
+                StaticTableSchema.builder(typeInfo.getTypeClass());
+
+        String[] fieldNames = typeInfo.getFieldNames();
+        for (int i = 0; i < typeInfo.getArity(); i++) {
+            TypeInformation<?> fieldType = typeInfo.getTypeAt(i);
+            int finalI = i;
+            addAttribute(
+                    tableSchemaBuilder,
+                    fieldNames[finalI],
+                    (attrT) -> attrT.getField(finalI),
+                    (TypeInformation<? super Object>) fieldType);
+        }
+
+        return tableSchemaBuilder.build();
+    }
+
+    private <attT> TableSchema<attT> createTableSchemaFromTuple(TupleTypeInfo<?> typeInfo) {
+        TypeInformation<?>[] fieldTypes = typeInfo.getFieldTypes();
+        String[] fieldNames = typeInfo.getFieldNames();
+
+        StaticTableSchema.Builder<?> tableSchemaBuilder =
+                StaticTableSchema.builder(typeInfo.getTypeClass());
+        for (int i = 0; i < fieldNames.length; i++) {
+            int finalI = i;
+            addAttribute(
+                    tableSchemaBuilder,
+                    fieldNames[finalI],
+                    (attrT) -> ((Tuple) attrT).getField(finalI),
+                    (TypeInformation<?>) fieldTypes[i]);
+        }
+
+        return (TableSchema<attT>) tableSchemaBuilder.build();
+    }
+
+    private <attrT> TableSchema<attrT> createTableSchemaFromPojo(PojoTypeInfo<attrT> typeInfo)
+            throws IntrospectionException {
+        StaticTableSchema.Builder<attrT> tableSchemaBuilder =
+                StaticTableSchema.builder(typeInfo.getTypeClass());
+        BeanInfo beanInfo = Introspector.getBeanInfo(typeInfo.getTypeClass());
+        PropertyDescriptor[] propertyDescriptors = beanInfo.getPropertyDescriptors();
```

Comment:
This looks like it might be expensive, can we cache the descriptors per type?

Reply:
I am not sure I get what we need to cache, these descriptors are only created once when creating the schema, during element conversion only the Enhanced getters are used.
Btw, this is also what the BeanTableSchema does :D